src/core_plugins/i18n/index.js

```diff
@@ -0,0 +1,7 @@
+export default function (kibana) {
+  return new kibana.Plugin({
+    init(server, options) {
+      // Add server routes and initalize the plugin here
```

Comment:
We'll need to register the translation registration API with the server here, much like the elasticsearch plugin does.